### PR TITLE
Adopt Contributor Covenant [GEOS-9295]

### PR DESCRIPTION
**Pull request by jodygarnett**
OSGeo changed its recommendations from its own home grown code of conduct to https://www.contributor-covenant.org

See https://www.osgeo.org/code_of_conduct/ for details.

This pull request was actually made from the GitHub "community" wizard for adding a code of conduct to your repository. Ideally I would like to update http://geoserver.org/comm/userlist-guidelines.html also.

<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] https://osgeo-org.atlassian.net/browse/GEOS-9295
- [x] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+when an individual is representing the project or its community. Examples of
+representing a project or community include using an official project e-mail
+address, posting via an official social media account, or acting as an appointed
+representative at an online or offline event. Representation of a project may be
+further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team at jody.. All
```
Comment (aaime) [inline CODE_OF_CONDUCT.md]: "contacting the project team at jody.." ?

Comment (jodygarnett) [reply CODE_OF_CONDUCT.md]: ha ha ha, stupid form.

So technically we should fill in Simone here as the project officer. I am tempted to fill in coc@osgeo.org instead.

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+when an individual is representing the project or its community. Examples of
+representing a project or community include using an official project e-mail
+address, posting via an official social media account, or acting as an appointed
+representative at an online or offline event. Representation of a project may be
+further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team via coc@osgeo.org. All
```
Comment (aaime) [inline CODE_OF_CONDUCT.md]: Much of the application of the covenant is based on interpretation, are all of the other projects making OSGeo set policy there?

Comment (jodygarnett) [reply CODE_OF_CONDUCT.md]: We can do anything we want, OSGeo only requires that we have a code of conduct. We can even use the old osgeo one here https://www.osgeo.org/resources/osgeo-code-of-conduct/

**Comment by aaime** [comment]
Well let's try it and see if it prevents honest discussion too, or not. +1

**Pull request merged**